pkg/cmd/server/origin/master.go

```diff
@@ -379,6 +381,37 @@ func handleVersion(req *restful.Request, resp *restful.Response) {
 	resp.ResponseWriter.Write(output)
 }
 
+// initOauthAuthorizationServerMetadataRoute initializes an HTTP endpoint for OAuth 2.0 Authorization Server Metadata discovery
+// https://tools.ietf.org/id/draft-ietf-oauth-discovery-04.html#rfc.section.2
+func initOauthAuthorizationServerMetadataRoute(container *restful.Container, path, masterPublicURL string) {
+	// Set up a service to return the OAuth metadata.
+	oauthWS := new(restful.WebService)
+	oauthWS.Path(path)
+	oauthWS.Doc("OAuth 2.0 Authorization Server Metadata")
+	oauthWS.Route(
+		oauthWS.GET("/").To(func(_ *restful.Request, resp *restful.Response) {
+			handleOauthAuthorizationServerMetadata(resp, masterPublicURL)
+		}).
+			Doc("get the server's OAuth 2.0 Authorization Server Metadata").
+			Operation("getOauthAuthorizationServerMetadata").
+			Produces(restful.MIME_JSON).
+			Consumes(restful.MIME_JSON))
```

Comment:
as long as that doesn't mean you have to pass `Content-Type: application/json` on the request, I don't care